### PR TITLE
Add image types, AMI IDs, extend AuthASG timeout for Cloudformation Enterprise

### PR DESCRIPTION
**Pull request by jon-can**
Added options for m4.large and m5.large. Added AMI IDs for all regions. Extended the timeout on the Auth ASG from 20 minutes to 30 minutes.

### CODE REVIEW

File: examples/aws/cloudformation/ent.yaml [hunk 1]
```diff
@@ -478,7 +492,7 @@ Resources:
     CreationPolicy:
       ResourceSignal:
         Count: 2
-        Timeout: PT20M
```
Comment (benarent) [inline examples/aws/cloudformation/ent.yaml]: We might want to consider running https://gravitational.com/teleport/docs/admin-guide/#troubleshooting and adding a health check to check is the system is ready using http://127.0.0.1:3000/healthz

**Review by benarent — APPROVED** [review]
Ok to test

File: examples/aws/cloudformation/ent.yaml (hunk 1 above)
Comment (webvictim) [reply examples/aws/cloudformation/ent.yaml:481]: There's actually some stuff in the startup scripts for Cloudformation specifically which does this - basically a command which sends the "OK I'm working" flag to Cloudformation at the end of the startup process. See https://github.com/gravitational/teleport/blob/master/examples/aws/cloudformation/ent.yaml#L229-L255

File: examples/aws/cloudformation/ent.yaml [hunk 2]
```diff
@@ -1001,4 +1015,4 @@ Outputs:
       Ref: Bucket
     Export:
       Name: S3BucketID
-
+
```
Comment (webvictim) [inline examples/aws/cloudformation/ent.yaml:1018]: Bonus tab here which isn't needed

File: examples/aws/cloudformation/ent.yaml [hunk 3]
```diff
@@ -13,6 +13,8 @@ Parameters:
     Default: t2.micro
     AllowedValues:
     - t2.micro
+    - m5.large
+    - m4.large
```
Comment (webvictim) [inline examples/aws/cloudformation/ent.yaml:17]: This is the full list of instance sizes we "support" from our marketplace listings:

m3.2xlarge
m3.large
m3.medium
m3.xlarge
m4.2xlarge
m4.4xlarge
m4.10xlarge
m4.16xlarge
m4.large
m4.xlarge
t2.large
t2.medium
t2.micro
t2.small
t2.xlarge

Might be worth adding all of these in as options.

**Review by webvictim — COMMENTED** [review]
Great start - can you also make the appropriate changes to `oss.yaml` in the same directory?

**Comment by webvictim** [comment]
ok to test

**Comment by benarent** [comment]
@jon-can Just doing a PR Sweep, would you be able to come back and review the extra comments.

**Comment by jon-can** [comment]
Sure I'll do this tomorrow.

**Comment by benarent** [comment]
just as another note @jon-can , we use our names for git branches. e.g. `jon-can/cloudformation-ami`

**Comment by russjones** [comment]
@jon-can Can you take a look at this and make the changes @webvictim suggested?

**Comment by benarent** [comment]
@jon-can  Just following up on this. Can you also make the appropriate changes to oss.yaml in the same directory?

**Comment by jon-can** [comment]
Yes I'll do that this evening.

**Review by webvictim — APPROVED** [review]
Approved! In future please put all changes in one PR.

**Comment by benarent** [comment]
ok to test

**Pull request merged**